### ISSUE TITLE
Error while running background job OCA\\Activity\\BackgroundJob\\RemoteActivity

### ISSUE THREAD

**Issue by Ornanovitch** [nextcloud/activity#1809]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

### Steps to reproduce
I don't really know

### Actual behaviour
We are spammed with this kind of error log from Activity:

```json
{
  "reqId": "6cCcCb1QGHR5WXKOaiOh",
  "level": 3,
  "time": "2024-10-12T09:40:02+00:00",
  "remoteAddr": "",
  "user": "--",
  "app": "core",
  "method": "",
  "url": "--",
  "message": "Error while running background job OCA\\Activity\\BackgroundJob\\RemoteActivity (id: 171689, arguments: [\"{masked}@{masked}\",\"PQk7SbxN0ZABNWu\",\"\\/{masked}\\/{file}.odt\",\"file_changed\",1728725880,\"{masked}@{masked}\"])",
  "userAgent": "--",
  "version": "29.0.8.1",
  "exception": {
    "Exception": "Error",
    "Message": "Typed property OCP\\BackgroundJob\\Job::$time must not be accessed before initialization",
    "Code": 0,
    "Trace": [
      {
        "file": "/var/www/nextcloud/lib/public/BackgroundJob/QueuedJob.php",
        "line": 61,
        "function": "start",
        "class": "OCP\\BackgroundJob\\Job",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/public/BackgroundJob/QueuedJob.php",
        "line": 47,
        "function": "start",
        "class": "OCP\\BackgroundJob\\QueuedJob",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/cron.php",
        "line": 177,
        "function": "execute",
        "class": "OCP\\BackgroundJob\\QueuedJob",
        "type": "->"
      }
    ],
    "File": "/var/www/nextcloud/lib/public/BackgroundJob/Job.php",
    "Line": 78,
    "message": "Error while running background job OCA\\Activity\\BackgroundJob\\RemoteActivity (id: 171689, arguments: [\"{masked}@{masked}\",\"PQk7SbxN0ZABNWu\",\"\\/{masked}\\/{file}.odt\",\"file_changed\",1728725880,\"{masked}@{masked}\"])",
    "exception": [],
    "CustomMessage": "Error while running background job OCA\\Activity\\BackgroundJob\\RemoteActivity (id: 171689, arguments: [\"{masked}@{masked}\",\"PQk7SbxN0ZABNWu\",\"\\/{masked}\\/{file}.odt\",\"file_changed\",1728725880,\"{masked}@{masked}\"])"
  },
  "id": "670a441cdd1c8"
}
```

### Server configuration

**Web server:** Nginx

**Database:** MariaDB

**PHP version:** 8.2

**Nextcloud version:** 29.0.8

<details>
<summary>List of activated apps</summary>

```
  - activity: 2.21.1
  - bookmarks: 14.2.6
  - calendar: 4.7.16
  - circles: 29.0.0-dev
  - cloud_federation_api: 1.12.0
  - collectives: 2.14.4
  - comments: 1.19.0
  - contacts: 6.0.0
  - dav: 1.30.1
  - deck: 1.13.2
  - event_update_notification: 2.4.0
  - federatedfilesharing: 1.19.0
  - files: 2.1.1
  - files_downloadlimit: 2.0.0
  - files_external: 1.21.0
  - files_pdfviewer: 2.10.0
  - files_reminders: 1.2.0
  - files_sharing: 1.21.0
  - files_trashbin: 1.19.0
  - files_versions: 1.22.0
  - groupfolders: 17.0.4
  - logreader: 2.14.0
  - lookup_server_connector: 1.17.0
  - music: 2.0.1
  - news: 25.0.0-alpha5
  - nextcloud_announcements: 1.18.0
  - notes: 4.11.0
  - notifications: 2.17.0
  - notify_push: 0.7.0
  - oauth2: 1.17.1
  - photos: 2.5.0
  - polls: 7.2.4
  - previewgenerator: 5.6.0
  - privacy: 1.13.0
  - provisioning_api: 1.19.0
  - related_resources: 1.4.0
  - richdocuments: 8.4.7
  - serverinfo: 1.19.0
  - settings: 1.12.0
  - sharebymail: 1.19.0
  - tables: 0.8.1
  - tasks: 0.16.1
  - text: 3.10.1
  - theming: 2.4.0
  - theming_customcss: 1.17.0
  - twofactor_backupcodes: 1.18.0
  - updatenotification: 1.19.1
  - user_oidc: 6.0.1
  - viewer: 2.3.0
  - workflowengine: 2.11.0

```
</details>

<details>
<summary>Nextcloud configuration</summary>

```
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "brume.felinn.org"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "29.0.8.1",
        "overwrite.cli.url": "https:\/\/brume.felinn.org",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "default_language": "fr",
        "default_locale": "fr_FR",
        "default_phone_region": "FR",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "forwarded_for_headers": [
            "HTTP_X_FORWARDED",
            "HTTP_FORWARDED_FOR",
            "HTTP_X_FORWARDED_FOR"
        ],
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379,
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "maintenance_window_start": 1,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "skeletondirectory": "\/etc\/nextcloud\/skeleton",
        "auth.webauthn.enabled": false,
        "preview_max_x": "1024",
        "preview_max_y": "1024",
        "jpeg_quality": "60",
        "blacklisted_files": [
            ".DS_Store"
        ],
        "theme": "default",
        "loglevel": 2,
        "updater.release.channel": "stable",
        "activity_use_cached_mountpoints": true,
        "app_install_overwrite": [
            "groupfolders",
            "files_lock",
            "news"
        ],
        "maintenance": false
    }
}

```
</details>

**Comment by Ornanovitch** [comment]
I realize this is only happening with activity from a federated user

**Comment by Pentaonia** [comment]
Same problem here, only when someone from an other NC has has added that and accesses files.

**Comment by joshtrichards** [comment]
Fixed in #1811
